modules/world/world.cpp

```diff
@@ -268,6 +268,18 @@ FrontRearAgents World::GetAgentFrontRearForId(
   return fr_agents;
 }
 
+void World::AddLabels(const LabelFunctions& label_evaluators) {
```

Comment:
Why do we add the Labels to the World and not to the Evaluator?
It clutters the world.

Reply:
Even though not implemented yet, the idea was that labels only have to be evaluated once for all evaluators, since some LTL rules are sharing the same labels. This should reduce the execution time, as labels do not have to be reevaluated. For this reason we would not want to store the labels in the evaluators. Another reason is, that LabelFunctions would also be directly passed to successor states of the world. This mechanism is currently used by the mvmcts-planner.

Reply:
however, if used and only used by the mvmcts planner, this seems to be a planner specific feature that is otherwise not needed.

I like the work you have done with the ltl-evaluators. However, I still do not see the necessity of these to be in the world.

I think this is a minor speed draw-back vs. a more cluttered implementation and more cloning effort (cloning the world in every step has an overhead on its own)

Reply:
And if a label describes the world state. Should this not be done maybe relative to a single vehicle. The the observed world would be a better place to add these labels.